AOV = Sum of total order amount / Total No. of orders 

In [1]:
import pandas as pd
import numpy as np 

In [2]:
spotify_data = pd.read_csv("2019 Winter Data Science Intern Challenge Data Set - Sheet1.csv")
spotify_data.head(10)

order_id  shop_id  user_id  order_amount  total_items payment_method  \
0         1       53      746           224            2           cash   
1         2       92      925            90            1           cash   
2         3       44      861           144            1           cash   
3         4       18      935           156            1    credit_card   
4         5       18      883           156            1    credit_card   
5         6       58      882           138            1    credit_card   
6         7       87      915           149            1           cash   
7         8       22      761           292            2           cash   
8         9       64      914           266            2          debit   
9        10       52      788           146            1    credit_card   

            created_at  
0  2017-03-13 12:36:56  
1  2017-03-03 17:38:52  
2   2017-03-14 4:23:56  
3  2017-03-26 12:43:37  
4   2017-03-01 4:35:11  
5  2017-03-14 15:25:01  
6  2017-03-01 21:37:57  
7   2017-03-08 2:05:38  
8  2017-03-17 20:56:50  
9  2017-03-30 21:08:26

In [3]:
spotify_data.dtypes

order_id           int64
shop_id            int64
user_id            int64
order_amount       int64
total_items        int64
payment_method    object
created_at        object
dtype: object

In [4]:
spotify_data.shape

(5000, 7)

In [5]:
spotify_data.isna().sum()

order_id          0
shop_id           0
user_id           0
order_amount      0
total_items       0
payment_method    0
created_at        0
dtype: int64

In [6]:
spotify_data.describe()

order_id      shop_id      user_id   order_amount  total_items
count  5000.000000  5000.000000  5000.000000    5000.000000   5000.00000
mean   2500.500000    50.078800   849.092400    3145.128000      8.78720
std    1443.520003    29.006118    87.798982   41282.539349    116.32032
min       1.000000     1.000000   607.000000      90.000000      1.00000
25%    1250.750000    24.000000   775.000000     163.000000      1.00000
50%    2500.500000    50.000000   849.000000     284.000000      2.00000
75%    3750.250000    75.000000   925.000000     390.000000      3.00000
max    5000.000000   100.000000   999.000000  704000.000000   2000.00000

order_amount = 3145.12 +/- 41282.53 

total_items = 8.78 +/- 116.32

There is a huge standard deviation relative to mean for both order_amount and total_items. From such large deviations, we can infer that there are outliers and large variance in the dataset. 
The standard error is quite high which signifies that the mean is likely an inaccurate representation of the true mean.

In [ ]:
# Evaluating the data (with outliers) using Interquartile Range statistic

In [7]:
# data points within IQR

q1 = spotify_data['order_amount'].quantile(0.25)
q3 = spotify_data['order_amount'].quantile(0.75)
q1, q3

(163.0, 390.0)

In [9]:
# inter-quartile range = difference between upper quartile and lower quartile 

iqr = q3 - q1
iqr

227.0

In [10]:
# data points lying above the upper limit are outliers 

upper_limit = q3 + 1.5*iqr
upper_limit

730.5

In [ ]:
# mean value when outliers are included : 3145.12

In [11]:
# mean value when outliers are excluded
spotify_data[(spotify_data['order_amount']<=upper_limit)]['order_amount'].mean()

293.7153735336489

In [ ]:
# median value when outliers are included: 284

In [12]:
# median value when outliers are excluded
spotify_data[(spotify_data['order_amount']<=upper_limit)]['order_amount'].median()

280.0

I would report the median as the metric. From the above results,we can observe that the median value including outliers is 284 and exclusing outliers is 280, which shows the median is more robust than mean against the outliers. 